### ISSUE TITLE
Crash on loot (spirit temple keys)

### ISSUE THREAD

**Issue by rattus128** [TestRunnerSRL/OoT-Randomizer#721]
Originally reported by hot bananas.

Reproduced on Biz and RA by original reporter.
Reproduced on Mupen Linux 2.5.9 by me.

This is very unlikely to be an emulator issue. Original report indicates it is specific to spirit temple keys.

```
  ":version":              "4.12.1 f.LUM",
  "file_hash":             ["Slingshot", "Stone of Agony", "Slingshot", "Stone of Agony", "Deku Stick"],
  ":seed":                 "QTH401ODI0",
  ":settings_string":      "AJTPL6MBB89LF9FAAJ3UHSE3LGP52XVZRBFQE3UGRU7593N4CNNJ",
```

To reproduce, walk to kak (wait for day) and talk to Anju for the loot.

**Comment by r0bd0g** [comment]
hot bananas Today at 9:27 PM
wait hold on
I think I narrowed it down
it only crashes if it's set to both "no logic" and "all locations reachable" is not checked

**Comment by mzxrules** [comment]
The problem is that these settings are causing the nes_message_data_static to fill completely. When this happens, message.py is coded to swap to writing dialogs into jpn_message_data_static, under the assumption that switching the segment offset bank from 07 to 08 will allow it to access the other file. Unfortunately, swapping banks does not swap the reference file used, and so when obtaining Spirit Temple Keys with this particular seed, it ends up reading data from 0192D000, which is in the middle of a compressed file, which isn't allowed

**Comment by sfla** [comment]
Experiencing the same, but with "Logic rules: Glitchless" (and "all_reachable: false").
```
  ":version":              "4.13.0 f.LUM",
  "file_hash":             ["Bottled Fish", "Master Sword", "Mushroom", "Skull Token", "Hover Boots"],
  ":seed":                 "R6ZEY9LVTB",
  ":settings_string":      "AJSCS23SA9VXU9CAAJAJGT9JUAA",
```
Beware; it's full entrance rando, but easy to reproduce:
1. Exit your house, you're in market.
2. Exit market north towards castle.
3. You're in Hyrule. Wait there until nightfall. 
4. Turn and go back inside to market
5. Enter treasure chest game, you're in guard house
6. Kill Gold Skulltula in crate.
7. Pick up spirit key and.. crash.

**Comment by flagrama** [comment]
As it depend on what text the rando has to inject, yeah it won't be consistent between every seed.

**Comment by jflatby** [comment]
Is there a way to detect whether the code has switched to bank 08 and has created some bad pointers on rom-generation?
And then just say rom generation failed, to avoid having to play several hours on an unbeatable rando before realizing again..

**Comment by mzxrules** [comment]
@jflatby It's possible to detect this yes, but there is no reason to ever force a failed generation here because we should have more than enough room for any dialog we wanted to add.

**Comment by mzxrules** [comment]
Possible fix #744

**Comment by TestRunnerSRL** [comment]
This should be fixed now
